### PR TITLE
More cookies support

### PR DESCRIPTION
**Pull request by devl00p**
Improve --cookie option to import existing cookies from Firefox or Chrome

Add --drop-cookies to reject cookies given by webservers

### CODE REVIEW

**Review by bretfourbe — COMMENTED** [review]
It would be nice to update the documentation for these changes and to make more explicit the fact that using `--cookie` option with `--drop-cookies` will prevent from cookies to be overwritten :)

**Comment by devl00p** [comment]
> It would be nice to update the documentation for these changes and to make more explicit the fact that using `--cookie` option with `--drop-cookies` will prevent from cookies to be overwritten :)

done

File: wapitiCore/main/wapiti.py [hunk 1]
```diff
@@ -1148,6 +1179,7 @@ def wapiti_main():
         if "output" in args:
             wap.set_output_file(args.output)
 
+        found_generator = False
```
[the remark this answers is not in the document]
Comment (devl00p) [reply wapitiCore/main/wapiti.py]: removed

File: wapitiCore/main/wapiti.py [hunk 2]
```diff
@@ -553,14 +554,33 @@ def add_excluded_url(self, url_or_pattern: str):
...
+            self.crawler.session_cookies = cookiejar
+        elif browser_name == "chrome":
+            cookiejar = browser_cookie3.chrome()
+            # There is a bug with this version of browser_cookie3 and we have to overwrite expiration date
+            # Upgrading to latest version gave more errors so let's keep an eye on future releases
+            for cookie in cookiejar:
+                cookie.expires = None
+            self.crawler.session_cookies = cookiejar
+        else:
+            raise InvalidOptionValue('--cookies', browser_name)
```
[the remark this answers is not in the document]
Comment (devl00p) [reply wapitiCore/main/wapiti.py]: indeed

File: wapitiCore/main/wapiti.py [hunk 3]
```diff
@@ -810,6 +830,12 @@ def wapiti_main():
         metavar="COOKIE_FILE"
     )
 
+    parser.add_argument(
+        "--drop-cookies",
+        action="store_true",
+        help=_("Refuse cookies from the webserver")
+    )
```
[the remark this answers is not in the document]
Comment (devl00p) [reply wapitiCore/main/wapiti.py]: done

**Comment by devl00p** [comment]
@bretfourbe I had to update some dependencies to fix test., `importlib_metadata` was the culprit once again. You should take notice of those changes to make sure it doesn't impact you.

**Pull request closed without merging**